client/src/hooks/system-status/useSystemOverview.ts

```diff
@@ -0,0 +1,97 @@
+//used in portal/client/src/workspace/SystemQueueTable.tsx
+//for getting system overview data (load%, running jobs, waiting jobs)
+import { useState, useEffect } from 'react';
+
+interface HPCSystem {
+  display_name: string;
+  hostname: string;
+  load_percentage: number;
+  is_operational: boolean;
+  in_maintenance: boolean;
+  jobs: {
+    running: number;
+    queued: number;
+  };
+}
+
+export function useSystemOverview() {
+  const [systems, setSystems] = useState<HPCSystem[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    async function fetchSystems() {
+      try {
+        setLoading(true);
+        const response = await fetch('/api/proxy/system-monitor/');
```

Comment:
Try creating a hook to handle this request with tanstack query. This will have the added benefit of loading state being built into the hook state.

Reply:
Just wanted to echo Sal's comment here